common/base-str.cpp

```diff
@@ -84,6 +84,24 @@ BASESTRING::BaseString(const BASESTRING &str)
 	assert(_str != nullptr);
 }
 
+
+TEMPLATE
+BASESTRING::BaseString(BASESTRING &&str) : _size(str._size) {
+	if (str.isStorageIntern()) {
+		// String in internal storage: copy it
+		memcpy(_storage, str._storage, _builtinCapacity * sizeof(value_type));
```

Comment:
That probably doesn't make much of a difference, but you only need to copy `_size + 1` values here (and in `assign()`) and do not need to copy the full storage size (`_builtinCapacity`).